### PR TITLE
adding node periodic jobs - kubetest2

### PR DESCRIPTION
**Pull request by Namanl2001**
Signed-off-by: Namanl2001 <namanlakhwani@gmail.com>

This PR is part of the CI migration from kubetest to kubetest2
https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing/2464-kubetest2-ci-migration

cc: @dims @spiffxp @amwat

### CODE REVIEW

**Review by amwat — COMMENTED** [review]
/lgtm

File: config/jobs/kubernetes/sig-node/node-docker.yaml [hunk 1]
```diff
@@ -28,6 +28,43 @@ periodics:
...
+      args:
+      - kubetest2
+      - noop
+      - --test=node
+      - --
+      - --repo-root=.
+      - --gcp-zone=us-west1-b
+      - --parallelism=8
+      - --focus-regex=\[NodeConformance\]
+      - --skip-regex=\[Flaky\]|\[Slow\]|\[Serial\]
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/node-docker.yaml:65]: comparing with `ci-docker-node-conformance`, why slow is also skipped?

File: config/jobs/kubernetes/sig-node/node-docker.yaml [hunk 2]
```diff
@@ -28,6 +28,43 @@ periodics:
   annotations:
     testgrid-dashboards: sig-node-docker
     testgrid-tab-name: docker-node-conformance
+- name: ci-docker-node-conformance-kubetest2
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/node-docker.yaml:31]: I'm curiuos if the provider=gce needs to be somehow specified or it is default?

File: config/jobs/kubernetes/sig-node/node-docker.yaml [hunk 3]
```diff
@@ -28,6 +28,43 @@ periodics:
...
+    timeout: 65m
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+  annotations:
+    testgrid-dashboards: sig-node-docker
+    testgrid-tab-name: docker-node-conformance-kubetest2
+  spec:
+    containers:
+    - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20211108-d0d3c6d7a8-experimental
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/node-docker.yaml:50]: requests and limits?

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml [hunk 4]
```diff
@@ -39,6 +39,55 @@ periodics:
...
+    base_ref: master
+    path_alias: k8s.io/test-infra
+  decoration_config:
+    timeout: 65m
+  annotations:
+    fork-per-release: "true"
+    fork-per-release-periodic-interval: 1h 2h 6h 24h
+    testgrid-dashboards: sig-release-master-blocking, sig-node-release-blocking
+    testgrid-tab-name: node-kubelet-master-kubetest2
+    testgrid-alert-email: kubernetes-sig-node+testgrid@googlegroups.com,kubernetes-sig-node-test-failures@googlegroups.com
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/node-kubelet.yaml:59]: perhaps we can start without e-mail notifications?

File: config/jobs/kubernetes/sig-node/node-docker.yaml (hunk 1 above)
Comment (Namanl2001) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:65]: Nice catch, it's unintenional

File: config/jobs/kubernetes/sig-node/node-docker.yaml (hunk 2 above)
Comment (Namanl2001) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:31]: yes, it's by default

File: config/jobs/kubernetes/sig-node/node-docker.yaml (hunk 3 above)
Comment (Namanl2001) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:50]: what type of requests and limits are you talking about? 🤔

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml (hunk 4 above)
Comment (Namanl2001) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml]: seems like tests need this alert to get passed. 

` Must have alert_options` ref: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/24308/pull-test-infra-unit-test/1459117935019692032

File: config/jobs/kubernetes/sig-node/node-docker.yaml (hunk 2 above)
Comment (SergeyKanzhelev) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:31]: thank you

File: config/jobs/kubernetes/sig-node/node-docker.yaml (hunk 3 above)
Comment (SergeyKanzhelev) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:50]: I'm not sure, I remember in one of previous PRs I saw the note that requests and limits will be needed on all containers (https://github.com/kubernetes/test-infra/pull/24039#issuecomment-962882732). but I don't know details

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml (hunk 4 above)
Comment (SergeyKanzhelev) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml]: sure, can we please only keep `kubernetes-sig-node-test-failures@googlegroups.com`, thank you

File: config/jobs/kubernetes/sig-node/node-docker.yaml (hunk 3 above)
Comment (Namanl2001) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:50]: Okay, so I didn't add them as there were no requests and limits present in the corresponding non-kubetest2 job.
I also don't know if it's needed.

cc: @ameukam @amwat

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

/hold

hold for the comment on limits and requests, feel free to unhold

File: config/jobs/kubernetes/sig-node/node-docker.yaml (hunk 3 above)
Comment (ameukam) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:50]: The requests and limits are only required for jobs running on `k8s-infra-prow-build*` (aka field `cluster: k8s-infra-prow-build`) clusters. They can be added now or when sig-node prowjobs will be migrated to the community-owned infrastructure.

File: config/jobs/kubernetes/sig-node/node-docker.yaml [hunk 5]
```diff
@@ -28,6 +28,43 @@ periodics:
   annotations:
     testgrid-dashboards: sig-node-docker
     testgrid-tab-name: docker-node-conformance
+- name: ci-docker-node-conformance-kubetest2
+  interval: 2h
+  decorate: true
+  path_alias: k8s.io/kubernetes
```
Comment (amwat) [inline config/jobs/kubernetes/sig-node/node-docker.yaml:34]: this should be replaced with a full ref under extra_refs

Comment (amwat) [reply config/jobs/kubernetes/sig-node/node-docker.yaml:34]:
```
- org: kubernetes
  repo: kubernetes
  base_ref: master
  path_alias: k8s.io/kubernetes
```

**Pull request merged**